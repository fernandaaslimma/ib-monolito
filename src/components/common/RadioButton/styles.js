import styled, { css } from "styled-components";
import { rem, media, remFontSize } from "../../../styles/tools";
import {
  grey40,
  grey70,
  lightestBlue20,
  white
} from "../../../styles/settings";

export const List = styled.ul`
  text-align: left;
  padding-left: ${rem(5)};
`;

export const RadioButtonTag = styled.label`
  display: inline-block;
  cursor: pointer;

  :hover {
    p {
      box-shadow: 0 ${rem(2)} ${rem(20)} ${lightestBlue20};
    }
  }

  input {
    position: absolute;
    visibility: hidden;
    pointer-events: none;

    :checked ~ p {
      background: ${lightestBlue20};
      color: ${white};
      box-shadow: 0 ${rem(2)} ${rem(20)} ${lightestBlue20};
    }
  }

  p {
    min-height: ${rem(42)};
    border-radius: ${rem(4)};
    border: solid ${lightestBlue20} ${rem(2)};
    display: flex;
    align-items: center;
    flex-wrap: nowrap;
    justify-content: center;
    margin-bottom: 5px;
    padding: ${rem(5)};
    color: ${lightestBlue20};
    transition: all 0.2s linear;
    ${media.md(css`
      min-height: ${rem(50)};
      padding: ${rem(5)};
    `)};
  }

  span {
    flex-grow: 1;
  }

  :active {
    position: relative;
    top: ${rem(2)};
  }

  ${media.md(css`
    margin-right: ${rem(8)};
    margin-left: ${rem(8)};
  `)};
`;

export const Item = styled.li`
  font-family: "Lato Semibold", "Lato";
  font-size: ${remFontSize(13)};
  font-style: italic;
  letter-spacing: ${rem(0.4)};
  color: ${grey70};
  padding: ${rem(6)} 0 ${rem(6)} ${rem(15)};
`;

export const Example = styled.span`
  font-family: "Lato";
  display: block;
  font-size: ${remFontSize(12)};
  font-weight: normal;
  letter-spacing: ${rem(0.3)};
  padding-top: ${rem(4)};
  color: ${grey40};
`;
