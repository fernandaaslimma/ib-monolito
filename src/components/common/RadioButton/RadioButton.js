import React from "react";
import { bool, string, func, oneOfType, array } from "prop-types";
import { RadioButtonTag } from "./styles";

function RadioButton({ label, onChange, checked, dataTest }) {
  const randomId = Math.floor(Math.random() * 10000);
  const labelIsArray = Array.isArray(label);
  return (
    <RadioButtonTag
      htmlFor={randomId}
      data-test={dataTest ? dataTest : `${randomId}_${RadioButton.displayName}`}
    >
      <input
        type="radio"
        id={randomId}
        checked={checked}
        onChange={e => onChange(e)}
      />
      <p>
        {labelIsArray
          ? label.map((label, index) => (
              <span key={`${label}_${index}`}>{label}</span>
            ))
          : `${label}`}
      </p>
    </RadioButtonTag>
  );
}

RadioButton.displayName = "RadioButton";

RadioButton.propTypes = {
  label: oneOfType([string, array]),
  onChange: func,
  checked: bool
};

export default RadioButton;
