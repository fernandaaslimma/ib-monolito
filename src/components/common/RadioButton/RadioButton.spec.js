import React from "react";
import { shallow, mount } from "enzyme";

import RadioButton from "./RadioButton";

describe("RadioButton component", () => {
  const props = {
    label: ["label 1", "label 2"],
    onChange: jest.fn()
  };

  it("should match snapshot", () => {
    expect(shallow(<RadioButton {...props} />)).toMatchSnapshot();
  });

  it("should change function", () => {
    const component = mount(<RadioButton {...props} />);
    component.find("input").prop("onChange")();

    expect(props.onChange).toHaveBeenCalled();
  });
});
