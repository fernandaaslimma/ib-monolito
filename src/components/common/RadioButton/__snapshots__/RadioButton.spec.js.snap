// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should match snapshot 1`] = `
<styled.label
  data-test="5000_RadioButton"
  htmlFor={5000}
>
  <input
    id={5000}
    onChange={[Function]}
    type="radio"
  />
  <p>
    <span
      key="label 1_0"
    >
      label 1
    </span>
    <span
      key="label 2_1"
    >
      label 2
    </span>
  </p>
</styled.label>
`;
